contracts/Conference.sol

```diff
@@ -103,6 +114,36 @@ contract Conference is Destructible, GroupAdmin {
 		participants[msg.sender] = Participant(_participant, msg.sender, false, false);
 	}
 
+	function unregisterInternal() internal {
+		// require(msg.value == deposit);
+		// require(isRegistered(msg.sender));
+
+		// Issue a refund
+
+		require(deposit > 0);
+		Participant participant = participants[msg.sender];
```

Comment:
I would retain a copy of the participant, and only transfer the deposit at the very end of the method when everything else has succeeded. Also, not sure if possible, but my coding instinct wants to break these up into private methods.